R/utils.R

```diff
@@ -210,6 +210,102 @@ get_currency_str <- function(currency,
   }
 }
 
+resolve_footnote_placement <- function(
+    data,
+    colname,
+    rownum,
+    input_placement,
+    cell_content,
+    context
+) {
+
+  if (input_placement %in% c("left", "right")) {
+    return(input_placement)
+  }
+
+  cell_alignment <-
+    get_alignment_at_body_cell(
+      data = data,
+      colname = colname,
+      rownum = rownum,
+      context = context
+    )
+
+  if (cell_alignment == "right") {
+    return("left")
+  } else {
+    return("right")
+  }
+}
+
+get_alignment_at_body_cell <- function(
+    data,
+    colname,
+    rownum,
+    context
+) {
+
+  column_alignment <-
+    dt_boxhead_get_alignment_by_var(
+      data = data,
+      var = colname
+    )
+
+  if (context != "html") {
+    return(column_alignment)
+  }
+
+  stub_layout <- get_stub_layout(data = data)
+
+  if (colname %in% stub_layout) {
+    column_alignment <- "right"
+  }
+
+  # Get cell alignment set by `tab_style`
+  styles_tbl <- dt_styles_get(data = data)
+
+  if (nrow(styles_tbl) > 0) {
+
+    styles_filtered_tbl <-
+      styles_tbl %>%
+      dplyr::filter(
+        locname == "data" &&
+          colname == .env$colname && rownum == .env$rownum
+      )
+
+    if (nrow(styles_tbl) < 1) {
+      return(column_alignment)
+    }
+
+    html_styles <-
+      vapply(
+        styles_filtered_tbl$styles,
+        FUN.VALUE = character(1),
+        USE.NAMES = FALSE,
+        FUN = styles_to_html
```

Comment:
I'm curious, why do you render the styles to html instead of looking at `styles_filtered_tbl$styles` directly?

Reply:
For a statement that's a bit more complicated like this: 

```r
exibble %>%
  gt() %>%
  fmt_number(num) %>%
  tab_footnote(footnote = "note", locations = cells_body(group, 1)) %>%
  tab_style(style = cell_text(align = "right"), locations = cells_body(group, 1)) %>%
  tab_style(style = cell_text(size = "smaller", align = "left"), locations = cells_body(group, 1)) %>%
  tab_style(style = "text-align: right; background: green;", locations = cells_body(group, 1))
```

The final alignment for the cell is `right` because `html_styles` will produce `"text-align: left; font-size: smaller; text-align: right; background: green;"`. That said, this code in the PR still gets this wrong because of the final `gsub()` statement.

I will rewrite this to use `styles_filtered_tbl$styles` because that is a better strategy anyway.

Reply:
Now rewritten, and many tests were added.